cpp/src/join/hash_join.cu

```diff
@@ -248,7 +248,8 @@ std::pair<rmm::device_vector<size_type>, rmm::device_vector<size_type>> probe_jo
   null_equality compare_nulls,
   cudaStream_t stream)
 {
-  size_type estimated_size = estimate_join_output_size<JoinKind, multimap_type>(
+  using estimate_size_type = int64_t;
+  size_type estimated_size = estimate_join_output_size<JoinKind, multimap_type, estimate_size_type>(
```

Comment:
Why do we need to pass estimate_size_type as a template parameter?

As per my observation, this is being used in one place, and even if we plan to use it somewhere else we would still use int64_t to evaluate and then check for overflow with expected type.

Reply:
I think you are right. I started out at the lower level and wanted to make it flexible, but then when I plumbed it up I  realized there is only one call site.  Will simplify.

Reply:
Done.